docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+* `FocusAction#prepare()` - Parse user command to suitable parameter for `PostponeAction#act()` function.
+
+Given below is an example usage scenario and how the focus mechanism behaves at each step.
+
+Step 1. The user executes `focus [task type]` command to filter based on task type. Command is then parsed by `FocusAction#prepare()` to be passed as arguments for `FocusAction#act()`.
+
+Step 2. `FocusAction#act()` then sets flag in the data.
+
+Step 3. Changed context is returned to inform the user through `Ui`.
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Postpone_Sequence_Diagram](Images/Focus.png)
+
+## Appendix A. Product scope
 ### Target user profile
 
-{Describe the target user profile}
+ - has a need to manage significant number of schedules 
+ - prefer desktop apps over other types 
+ - can type fast
+ - prefers typing to mouse interactions 
+ - is reasonably comfortable using CLI apps 
 
 ### Value proposition
+All in one app to track tasks and their dates, monitor productivity and calculate cap. 
 
-{Describe the value proposition: what problem does it solve?}
-
-## User Stories
+## Appendix B. User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
+|Priority| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| *** |Student before start of semester|List the modules MC|Follow the recommended MC |
+| *** |Student before start of semester|List of module available|Easily Choose which modules to take|
+| ** |Student before start of semester|Find the modules either by keyword, module code or even MC |Easily see the desired modules |
+| ** |Student before start of semester|Select the modules but not taking it yet |Easily whether the MC fits my requirement |
+| *  |Student before start of semester|Find out the etails of the Module|To find out more about the modules.|
+| *** |Student before start of semester|Take the desired modules|Mark the modules that i want to take as taken  |
+| *** |Student during the semester|Add tasks such as todo,deadline and event into my list|Easily keep track of all the task i have to complete  |
+| ** |Student during the semester|Have a Reminder of which deadline is due soon|Ensure that no task is missed out  |
+| *** |Student during the semester|Add task to modules |Easily know which tasks belongs to which modules  |
+| *** |Student during the semester|Delete task once they are completed |Remove unnecessary task on the list |
+| ** |Student after the semester|Calculate the CAP of my individual modules|Easily find out my performance this semester |
+| * |Student after the semester|Clear the list of tasks and modules|Start afresh for the next semester |
```

Comment:
I can see that currently you split the user stories into three groups which is good. But as you have already prioritized them, I suggest that you could reorder them by their priorities.
For example:
*** before
**  before
*** during
** during
...